src/setupProxy.js

```diff
@@ -24,14 +24,27 @@ function setupTestProxy(app) {
 
   app.use((req, res, next) => {
     const parts = url.parse(req.url);
-    const integrateQuery = parts.pathname + (parts.search ? encodeURIComponent(parts.search) : '');
-    const filePath = path.join(__dirname, 'test/fixtures', integrateQuery);
-
-    if (fs.existsSync(filePath)) {
-      fs.readFile(filePath, 'utf8', (err, contents) => {
-        res.setHeader('Content-Type', 'application/json');
-        res.send(contents);
-      });
+    const filePath = path.join(__dirname, 'test/fixtures', parts.pathname);
+    const queryFilePath = path.join(filePath, parts.search ? encodeURIComponent(parts.search) : '');
+    const indexFilePath = path.join(filePath, 'index.html');
+
+    const sendFile = path => fs.readFile(path, (err, contents) => {
+      res.end(contents);
+    });
+
+    const isFile = path =>
+      fs.existsSync(path)
+      && fs.existsSync(fs.realpathSync(path))
+      && fs.lstatSync(fs.realpathSync(path)).isFile();
+
+    const isDirectory = path => fs.existsSync(path) && fs.lstatSync(path).isDirectory();
+
+    if (isFile(queryFilePath)) {
+      sendFile(queryFilePath);
```

Comment:
When do we serve filenames with the query string in them? I did not see any of them in the diff

Reply:
osweb queries, this logic was already here i just moved it